src/parser/priest/shadow/modules/spells/azeritetraits/WhispersOfTheDamned.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS/index';
+import { calculateAzeriteEffects } from 'common/stats';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import TraitStatisticBox, { STATISTIC_ORDER } from 'interface/others/TraitStatisticBox';
+import ItemDamageDone from 'interface/others/ItemDamageDone';
+import Events from 'parser/core/Events';
+import HIT_TYPES from 'game/HIT_TYPES';
+import { formatNumber } from 'common/format';
+
+const whispersOfTheDamnedStats = traits => Object.values(traits).reduce((obj, rank) => {
+  const [damage] = calculateAzeriteEffects(SPELLS.WHISPERS_OF_THE_DAMNED.id, rank);
+  obj.damage += damage;
+  return obj;
+}, {
+  damage: 0,
+});
+
+const MIND_BLAST_SPELLS = [
+  SPELLS.MIND_BLAST,
+  SPELLS.SHADOW_WORD_VOID_TALENT,
+];
+
+/**
+ * Whispers of the Damned
+ * Mind Blast deals 2704 additional damage and generates 20 additional Insanity on critical strikes.
+ *
+ * Example log: /report/JQNwLbpdtmrzYAGC/5-Mythic+Blackwater+Behemoth+-+Kill+(5:09)/Adoraci
+ */
+class WhispersOfTheDamned extends Analyzer {
+  damageValue = 0;
+  damageDone = 0;
+  insanityGained = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.WHISPERS_OF_THE_DAMNED.id);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(MIND_BLAST_SPELLS), this.onDamageEvent);
+    if (!this.active) {
+      return;
+    }
+
+    const { damage } = whispersOfTheDamnedStats(this.selectedCombatant.traitsBySpellId[SPELLS.WHISPERS_OF_THE_DAMNED.id]);
+    this.damageValue = damage;
+  }
+
+  onDamageEvent(event) {
+    this.damageDone += this.damageValue;
+
+    if (event.hitType === HIT_TYPES.CRIT) {
+      this.insanityGained += 20;
```

Comment:
This will be misleading if the spell overcaps.

Reply:
Updated to use an energize event and `resourceChange` to get exactly how much insanity was gained from the trait procing